Makefile

```diff
@@ -116,8 +118,14 @@ endif
 	go generate ./pkg/... ./cmd/...
 	./scripts/sync-up-helm-chart.sh
 
-push:
-	$(DOCKER) push $(REGISTRY)/$(TARGET):$(GIT_VERSION)
+push: container
+	docker login -u "$(QUAY_USERNAME)" -p "$(QUAY_PASSWORD)" quay.io
```

Comment:
Two issues here:

1. We should use the global `$(DOCKER)` variable.

2. This will be executed by users that may not be using quay and won't have these user/pass variables defined. I suggest passing a flag from `.travis.yml` when invoking `make push` and adding a check for that flag before running this `docker login` command. Or, you can create a new make target such as `travis_login` that is invoked by `.travis.yml` before invoking `make push`.

Reply:
Added a new target `travis_login`, but seems there is no `QUAY_USERNAME` and `QUAY_PASSWORD` env params.

```
$ test $DOC_ONLY && echo "Skipped!" || make travis_login && make push
docker login -u "" -p "" quay.io
Username: 
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
The build has been terminated
```
@font  Could you help take a look? Thank you!

Reply:
@qinpingli That's expected as those credential environment variables are only defined whenever travis runs an official build from master or a new tagged release of federation. So we only need to run the `docker login` command when we need to push to the official federation registry repo. For example from the `imagebuild.sh` script, we could invoke the new `travis_login` target (or rename to `travis_image_build` or similar), but within it we would need to check `TRAVIS_TAG` and `TRAVIS_BRANCH` exists and match `master` or a tagged release version to see if we need to login and push, otherwise we don't bother.